### DISCUSSION TITLE
ScriptHandler Pluggable to permit use FEEL Expression

### DISCUSSION THREAD

**Discussion by ingpconci** [bpmnServer/bpmn-server#224] [Ideas]
In the **bpmn-server version 1** the script handler was pluggable.
I've implemented:
appDelegate.scopeEval -->evaluateExpression
appDelegate.scopeJS   -->executeScript
to permit the processing of the FEEL expression write in the BPMN file using the Camunda Modeler
![image](https://github.com/user-attachments/assets/2dbd1c2b-92e8-40d8-81d7-918ba2c7445e)

`
scopeEval(scope, script: string) {
        debug('MyAppDelegate.scopeEval: Starting... scope=', scope, ' script=', script);
        let result;
        // remove the = from the init of the script
        // https://docs.camunda.io/docs/components/concepts/expressions/#expressions-vs-static-values
        let scriptModif1;
        if (script.startsWith("=")) {
            scriptModif1 = script.substring(1);
        } else {
            scriptModif1 = script;
        }
        // duplicate internal = in the script to permit use ELF espression
        debug('MyAppDelegate.scopeEval: step1 removed = from init; scriptModif1: ', scriptModif1);

        let scriptModif2;
        if (scriptModif1.indexOf("=")) {
            scriptModif2 = scriptModif1.replace("=", "==");
        }
        else {
            scriptModif2 = scriptModif1;
        }
        debug('MyAppDelegate.scopeEval: step2 duplicate = inside the script; scriptModif2: ', scriptModif2);
`

Is it possible to reinsert the possibility of customizing the script handler?
It is possible to use the the javascript-feel-engine as scriptHandler?
https://docs.camunda.io/docs/components/modeler/feel/what-is-feel/#feelin-javascript-feel-engine

**Comment by bpmn-ts** [comment]
Hi
Yes, it is possible, I am working on it
Thanks

**Reply by ingpconci** [reply]
Hi
Can I help modify the ScriptHandler? I need to update my application with bpmn-server ver 2, but without that modification, my BPMN process doesn't run.

**Comment by ralphhanna** [comment]
New Release coming shortly
Thanks
________________________________

**Reply by ralphhanna** [reply]
New release is now available, please have a look and let me know
